Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -101,13 +106,36 @@ internal class ConnectionPool {
         }
     }
 
+    /// Execute the provided closure with the pointer to the created `mongoc_client_pool_t`, potentially waiting
+    /// for the pool to finish starting up first. The closure will be executed while the stateLock is held, so
+    /// closure MUST NOT attempt to acquire the lock.
+    ///
+    /// Throws an error if the pool is closed or closing.
+    private func withResolvedPool<T>(body: (OpaquePointer) throws -> T) throws -> T {
+        try self.stateLock.withLock {
+            switch self.state {
+            case let .open(pool):
+                return try body(pool)
+            case let .opening(future):
+                let pool = try future.wait()
```

Comment:
is it possible that we encounter a race condition where a user creates a client and immediately fires off [threadPoolSize] operations, those somehow manage to get scheduled on the thread pool before the pool creation does, and then all of those are stuck blocking on this future which can't complete? 🤔

Reply:
Assuming the thread pool operates on a first-come-first-serve basis, the pool creation should always get scheduled first since it happens before the `ConnectionPool` initializer even returns. If there aren't any ordering guarantees then it seems that type of race could be possible.

From looking at the implementation of `runIfActive`, the pool acquires a lock and then appends the work item to a queue and then returns. It looks like the actual worker threads then take work items from the front of the queue, so it seems the first-come-first-serve ordering is preserved here. Good idea though, this could have been a serious issue if the pool wasn't ordered.